src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,264 @@
+    #  - OR use this?: get_all_owned_terms()
+    #    - it seems long and messy compared to these 3 lines, but is more comprehensive. I could make
+    #      get_all_owned_terms() a simple 3 line func and then make a separate func for all edge cases?
+    #  - note I may have list(owned_prefix_map.keys())[0] instead of list(owned_prefix_map)[0] ins ome places
+    # - filter by source
+    mappings_df['source_prefix'] = mappings_df['source_id'].apply(lambda x: x.split(':')[0])
+    mappings_df = mappings_df[mappings_df['source_prefix'].isin(owned_prefix_map.keys())]
+    del mappings_df['source_prefix']
+    # - filter obsoletes (from mappings_df --> source_df & mondo_df)
+    #  - filter obsoleted in Mondo
+    mappings_df = mappings_df[~mappings_df['mondo_label'].str.startswith('obsolete')]
+    #  - filter obsoleted in source
+    source_obsoletes: Set[CURIE] = set([x for x in source_db.obsoletes()])
+    mappings_df = mappings_df[~mappings_df['source_id'].isin(source_obsoletes)]
+    # - filter mapped but deleted IDs
+    #   If there is no label, we take that as a proxy / sign that it has been deleted. Alternative ways: entities(),
+    #   all_entity_curies(), node() (empty props), node_exists() (no: falsely says 'True', prolly cuz bad xrefs exist)
+    source_labels: List[Tuple[CURIE, str]] = [x for x in source_db.labels(mappings_df['source_id'].tolist())]
+    source_deleted_ids: Set[CURIE] = set([x[0] for x in source_labels if not x[1]])
+    mappings_df = mappings_df[~mappings_df['source_id'].isin(source_deleted_ids)]
+    if len(mappings_df) == 0:
+        logging.warning(f'Synonym Sync: No mappings found for source {source_name}. Exiting.')
+        return
+    # - update source labels (sometimes Mondo & source have discrepancy)
+    source_labels: Dict[CURIE, str] = {curie: label for curie, label in source_labels}
+    mappings_df['source_label'] = mappings_df['source_id'].map(source_labels)
+
+    # Query synonyms: source
+    source_df: pd.DataFrame = _query_synonyms(mappings_df['source_id'].tolist(), source_db)
+    # - add mapped mondo ids
+    # TODO: drop mondo_id from source_df
+    # source_mondo_lookup = mappings_df.set_index('source_id')['mondo_id'].to_dict()
+    # source_df['mondo_id'] = source_df['curie'].map(source_mondo_lookup)
+    # source_df = source_df.rename(columns={'curie': 'source_id'})[['mondo_id', 'source_id', 'synonym_scope', 'synonym']]
+    source_df = source_df.rename(columns={'curie': 'source_id'})[['source_id', 'synonym_scope', 'synonym']]
+
+    # Query synonyms: Mondo
+    mondo_df = pd.read_csv(mondo_synonyms_path, sep='\t').rename(columns={'?dbXref': 'source_id'}).fillna('')
+    mondo_df = mondo_df.rename(columns=lambda x: x.lstrip('?'))
+    mondo_evidence_lookup: Dict[Tuple, List[str]] = mondo_df.groupby(['mondo_id', 'synonym_scope', 'synonym'])['source_id'].agg(list).to_dict()
+    mondo_evidence_lookup: Dict[Tuple, str] = {k: ', '.join(v) for k, v in mondo_evidence_lookup.items()}
```

Comment:
## Evidence column

Reply:
> Delimitation: ,  ok, or prefer , (no space), or even |?

In robot templates we generally prefer `|`

> ORCIDs are URIs. I could change them to CURIEs? Does it matter?

It does not matter (your call), but for brevity, why not. `orcid:123...`

> Evidence column only should appear in -updated template. Correct?

Do you mean the evidence from Mondo (not source ontology)? If so, then yes, updated is enough. "added" cannot have evidence by definition, and for "confirmed" we dont need Mondo evidence.

> Included all evidence in the column

Yeah, for now just put all of it.

Reply:
@matentzn 
> In robot templates we generally prefer `|`

You asked for commas! I shoulda spoke up lol.

> Do you mean the evidence from Mondo (not source ontology)? I
Yes. Do you want me to rename the column `mondo_evidence`?

Reply:
> You asked for commas! I shoulda spoke up lol.

And I now remember that -> The reason is that comma is a bit more readable, and in this case the column is not actionable (i.e. does not translate into annotation axioms in the ROBOT template). Since it is only there to aid curators during personal review, it does not matter which symbol you use for the split. Sorry about forgetting about that.

> Yes. Do you want me to rename the column mondo_evidence?

Sounds great!

Reply:
Done~ Marking resolved.